app/controllers/projects/samples/attachments_controller.rb

```diff
@@ -27,13 +27,33 @@ def create
         end
       end
 
-      def destroy
-        authorize! @project, to: :update_sample?
+      def destroy # rubocop:disable Metrics/MethodLength, Metrics/AbcSize
+        authorize! @project, to: :destroy_sample?
+        respond_to do |format|
+          if @attachment.attachable_type != 'Sample' || @attachment.attachable_id != @sample.id
```

Comment:
I'm not sure how this if condition could ever be hit. The reason being that `@attachment` is queried from the `@sample`, so it should only be set if the attachment belongs to the sample. There is also no test for this condition, so that would be required if this if statement is possible to be met which I cannot see it ever happening. If the user were trying to delete an attachment that doesn't belong to the sample they would get `not_found` as defined within the `attachment` method in this controller.

Reply:
Following up, this is what happens when I change the attachment id in the `delete` form button to point to an attachment that belongs to another Sample.

![image](https://github.com/phac-nml/irida-next/assets/492127/67d3a655-076c-4c7e-90a9-226147d70c8b)

Which confirms my comments that this condition can never be met.

Reply:
Removed in ffb0a839fee97a3e3eb2bf7596e329fc451758ee